Gems/ROS2/Code/Source/RobotImporter/URDF/VisualsMaker.cpp

```diff
@@ -225,36 +226,262 @@ namespace ROS2
+    {
+        auto material = visual->Material();
+
+        if (!material)
+        {
+            // Material is optional, it might not appear on all visuals.
             return;
         }
 
-        AZ::Entity* entity = AzToolsFramework::GetEntityById(entityId);
+        AZ_Assert(material->Element(), "Material has data but no Element pointer. Something unexpected has happened with the SDF parsing.");
 
-        // As a Material doesn't have a name and there can only be 1 material per <visual> tag,
-        // the Visual Name is used for the material
-        const std::string materialName{ visual->Name() };
-        const AZStd::string azMaterialName{ materialName.c_str(), materialName.size() };
+        // Conversions from <material> in the file to O3DE are extremely imprecise because the data is going from a Phong model to PBR,
+        // and there are no direct translations from one type of lighting model to the other. All of the conversions will create some
+        // rough approximations of the source lighting data, but should hopefully provide a reasonable starting point for tuning the look.
 
-        // If present in map, take map color definition as priority, otherwise apply local node definition
-        const auto materialColorUrdf =
-            m_materials.contains(azMaterialName) ? m_materials.at(azMaterialName)->Diffuse() : visual->Material()->Diffuse();
-        const AZ::Color materialColor = URDF::TypeConversions::ConvertColor(materialColorUrdf);
+        // Also, URDF/SDF files don't have a concept of overriding specific materials, so every material override generated
+        // below will get applied to *all* materials for a mesh file.
 
-        entity->CreateComponent(AZ::Render::EditorMaterialComponentTypeId);
-        AZ_Trace("AddVisual", "Setting color for material %s\n", azMaterialName.c_str());
-        entity->Activate();
-        AZ::Render::MaterialComponentRequestBus::Event(
-            entityId,
-            &AZ::Render::MaterialComponentRequestBus::Events::SetPropertyValue,
-            AZ::Render::DefaultMaterialAssignmentId,
-            "settings.color",
-            AZStd::any(materialColor));
-        entity->Deactivate();
+        // First, force the model asset to get loaded into memory before adding the material component.
+        // This is required so that we can get the default material map that will be used to override properties for each material.
+        auto modelAsset = AZ::Data::AssetManager::Instance().GetAsset<AZ::RPI::ModelAsset>(assetId, AZ::Data::AssetLoadBehavior::Default);
+        modelAsset.BlockUntilLoadComplete();
+
+        // Initialize the material component configuration to contain all of the material mappings from the model.
+        AZ::Render::MaterialComponentConfig config;
+        config.m_materials = AZ::Render::GetDefautMaterialMapFromModelAsset(modelAsset);
```

Comment:
@gadams3 Just fixed the name of thus function to be `GetDefaultMaterialMapFromModelAsset` and the old name is now deprecated.

Reply:
He fixed it in stabilization, so there's a timing issue where it isn't fixed in development yet.

Reply:
Should we wait with merging this until the the aforementioned change is in o3de development? The alternative is to target stabilization2310 branch of o3de-extras now with this PR.

Reply:
@adamdbrw the code builds with o3de `development` only due to changes in `SceneApi`, so targeting `stabilization/2310` in o3de-extras would be incorrect

Reply:
The spelling change should get pulled into development today I think, so I'll wait to submit this until after it's got the correct spelling.

Reply:
Spelling change is in dev, so I've now changed it here and I'm running AR.